### DISCUSSION TITLE
toppings & options

### DISCUSSION THREAD

**Discussion by lastraum** [RIAEvangelist/node-dominos-pizza-api#95] [Q&A] [answered]
Excited for v3.x soon! However, I would like to know how to understand what the codes are for the toppings and options so that i can "build my own" pizza.

**Comment by RIAEvangelist** [comment, marked as answer]
Here's an example from v3, options format has not changed though.

```js
import {Item} from 'dominos';

const cheesePizza=new Item(
    {
        ID:1,
        code:'14SCREEN',
        options:{
            // this says, for the whole pizza put sauce on it
            X: {'1/1' : '1'}, 
            
            //this says for the whole pizza put double cheese
            C: {'1/1' : '2'}
        }
    }
)
```

if you `console.dir` out the products your are interested you will see the options string in there. you just have to sort of poke around. I never got too deep into them. If you want to look into it and learn more, I would take any documentation, code samples or whatever.

```js 
console.dir(menu.menu..., {depth:3});
```

**Reply by lastraum** [reply]
Thanks for the quick reply. So there's no reference to what each code is? 

X = sauce
C = cheese

Etc

**Comment by RIAEvangelist** [comment]
Sadly not yet. You can look at the menu and sort of figure it out because
many items have the defaults written out.

Also... If you wanted too... You could go to the website, look at the
requests as you change the toppings and make a note... Or even just make
some basic notes from the obvious ones in the menu, like P is pepperoni.

If you were to write a table or even just a list in an Options markdown, I
would totally merge it into v3 and people would forever love you and your
delicious opensource contribution.

Insert image of sad puppy and a multitude of obscenely cute kittens here...

**Comment by RIAEvangelist** [comment]
Even if you just wrote a for loop to process all of the options in the returned menu object, that would be a big help

**Comment by RIAEvangelist** [comment]
I found a URL that can help figure this out.

https://order.dominos.com/power/store/8278/menu?lang=en

I removed the structured arg from the end and got an unstructured data dump.

**Comment by RIAEvangelist** [comment]
Also, I just realized that all of the options exist on the `.toppings` object of the menu. You can have a look in the [docs/Menu.md](https://github.com/RIAEvangelist/node-dominos-pizza-api/blob/master/docs/Menu.md) for anyone reading this, try logging it out to explore more.
